packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterItem.tsx

```diff
@@ -0,0 +1,279 @@
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+            break;
+
+          default:
+            if (
+              event.metaKey ||
+              event.ctrlKey ||
+              event.altKey ||
+              event.shiftKey ||
+              event.key === 'Tab'
+            ) {
+              break;
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+        }
+      },
+      [apiRef, colDef.field, colIndex, columnFields, headerFilterMenuRef, isEditing, isMenuOpen],
+    );
+
+    const publish = React.useCallback(
+      (eventName: keyof GridHeaderFilterEventLookup, propHandler?: React.EventHandler<any>) =>
+        (event: React.SyntheticEvent) => {
+          apiRef.current.publishEvent(
+            eventName,
+            apiRef.current.getColumnHeaderParams(colDef.field),
+            event as any,
+          );
+          if (propHandler) {
+            propHandler(event);
+          }
+        },
+      [apiRef, colDef.field],
+    );
+
+    const onClick = React.useCallback(
+      (event: React.MouseEvent) => {
+        if (!hasFocus) {
+          apiRef.current.startHeaderFilterEditMode(colDef.field);
+          apiRef.current.setColumnHeaderFilterFocus(colDef.field, event);
+        }
+      },
+      [apiRef, colDef.field, hasFocus],
+    );
+
+    const mouseEventsHandlers = React.useMemo(
+      () => ({
+        onKeyDown: publish('headerFilterKeyDown', onKeyDown),
+        onClick: publish('headerFilterClick', onClick),
+      }),
+      [onClick, onKeyDown, publish],
+    );
+
+    const ownerState = {
+      ...rootProps,
+      colDef,
+    };
+
+    const classes = useUtilityClasses(ownerState as OwnerState);
+
+    const isNoInputOperator = NO_INPUT_OPERATORS[colDef.type!]?.includes(item.operator);
+    const isFilterActive = hasFocus || Boolean(item?.value) || isNoInputOperator;
+
+    const placeholder = apiRef.current.getLocaleText('columnMenuFilter');
+
+    return (
+      <div
+        className={clsx(classes.root, headerClassName)}
+        ref={handleRef}
+        style={{
+          height,
+          width,
+          minWidth: width,
+          maxWidth: width,
+        }}
+        role="columnheader"
+        aria-colindex={colIndex + 1}
+        aria-label={headerFilterComponent == null ? colDef.headerName ?? colDef.field : undefined}
+        {...other}
+        {...mouseEventsHandlers}
+      >
+        {headerFilterComponent}
+        {InputComponent && !headerFilterComponent ? (
+          <InputComponent
+            apiRef={apiRef}
+            item={item}
+            inputRef={inputRef}
+            applyValue={applyFilterChanges}
+            onFocus={() => {
+              apiRef.current.startHeaderFilterEditMode(colDef.field);
+            }}
+            onBlur={() => {
+              apiRef.current.stopHeaderFilterEditMode();
+            }}
+            fullWidth
+            placeholder={placeholder}
```

Comment:
I think you can directly inline it.

```suggestion
            placeholder={apiRef.current.getLocaleText('columnMenuFilter')}
```